tests/msc2403_test.go

```diff
@@ -426,5 +465,13 @@ func publishAndCheckRoomJoinRule(t *testing.T, c *client.CSAPI, roomID, expected
 	if !roomFound {
 		t.Fatalf("Room was not present in public room directory response")
 	}
+}
 
+// TestCannotSendNonKnockViaSendKnock checks that we cannot submit anything via /send_knock except a knock
+func TestCannotSendNonKnockViaSendKnock(t *testing.T) {
+	testValidationForSendMembershipEndpoint(t, "knock", "/_matrix/federation/v2/send_leave",
```

Comment:
It looks like the expected membership and the path aren't passed in the right order.